src/workbench.js

```diff
@@ -52,32 +53,107 @@ const onlyWithFeature = (featureName, component) =>
  * The React components will be created without any props. If you need the
  * components to have props, use a wrapper HOC.
  */
-const componentRegistry = {
-  'beacon-list': onlyWithFeature('beacons', views.BeaconList),
-  'connection-list': views.ConnectionList,
-  'dataset-list': views.DatasetList,
-  'dock-list': onlyWithFeature('docks', views.DockList),
-  'feature-list': onlyWithFeature('features', views.FeatureList),
-  'ground-control-view': injectFlockFromContext(views.GroundControlView),
-  'layer-list': views.LayerList,
-  'light-control': onlyWithFeature('showControl', views.LightControlPanel),
-  'lcd-clock-panel': views.LCDClockPanel,
-  'log-panel': views.LogPanel,
-  map: MapView,
-  messages: views.MessagesPanelView, // deprecated, kept there for compatibility
-  placeholder: Nothing,
-  'saved-location-list': views.SavedLocationList,
-  'show-control': onlyWithFeature('showControl', views.ShowControlPanel),
-  'three-d-view': onlyWithFeature('threeDView', views.ThreeDTopLevelView),
-  'uav-list': injectFlockFromContext(views.UAVList),
+
+export const componentRegistry = {
+  'beacon-list': {
+    component: views.BeaconList,
+    label: 'Beacons',
+    detachable: true,
+    feature: 'beacons',
+  },
+  'connection-list': {
+    component: views.ConnectionList,
+    label: 'Connections',
+  },
+  'dataset-list': {
```

Comment:
Not used yet, but you can safely make this detachable as well; it's a plain list-of-items just like the beacon list.

Reply:
Actually, on second thought, let's mark it as deprecated just like we did with `messages`. I know it's not related to the feature being implemented in this PR, but let's do it anyway now that we are tidying up this part of the code.